product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+---
+title: Performing a Postgres major version rolling upgrade on a PGD cluster built with TPA
+navTitle: Upgrade Postgres major version
+deepToC: true
+---
+
+## Overview
+
+You can perform a major version upgrade of Postgres on a Postgres Distributed (PGD) cluster deployed by Trusted Postgres Architect (TPA).
+
+The rolling upgrade process allows updating individual cluster nodes to a new major Postgres version while maintaining cluster availability and operational continuity. This approach minimizes downtime and ensures data integrity by allowing the rest of the cluster to remain operational as each node is upgraded sequentially. 
+
+We recommend upgrading your Postgres cluster to a newer major version to access improved features, performance enhancements, and security updates. These step-by-step instructions and a worked example help to provide a smooth and controlled upgrade process.
+
+### Plan the upgrade
+
+The cluster must be available throughout the process (that is, a *rolling* upgrade). To achieve continuous cluster availability, you must [perform a planned switchover](#performing-a-planned-switchover) of the [write leader](https://www.enterprisedb.com/docs/pgd/latest/terminology/#write-leader) node before upgrading it so that a write leader is always available. 
+
+You can see which node is the write leader by connecting to the PGD cluster (for example, using SSH) and running the `pgd show-groups` command:
+
+```bash
+sudo -u postgres pgd show-groups
+```
+
+To keep the number of planned [switchovers](https://www.enterprisedb.com/docs/pgd/latest/cli/command_ref/pgd_switchover/) to a minimum, upgrade the current write leader last in your upgrade order, so you have to perform only [one planned switchover](#performing-a-planned-switchover) during the upgrade process.
+
+### Perform the upgrade on each node
+
+!!! Note
+To help prevent data loss, ensure that your databases and configuration files are backed up before starting the upgrade process.
+!!!
+
+Next, perform the following steps on each node in order:
+
+1. If the target node is the current write leader, perform a [planned switchover](#plan-the-upgrade).
```

Comment:
Ok, couple of big comments here. This is the meat of the article, and it's one big step. And, what were numbered steps are not anymore because when you added notes to that broke the sequential count in markdown. Split them out into headings and make it sync with the worked example below.

Reply:
Split the steps up into headings as opposed to using numbers? I am not 100% on what you are suggesting here.

Reply:
Go look at the deployment.

Every step is numbered 1.

And there's a lot of steps.

Break them up.

Put headings on them.

Make the headings match the worked example's headings.

Reply:
1. I see that the numbering isn't working; I could fix that by indenting the note under each by one, but it looks like you are suggesting to do away with the ordered list. 
2. When you say break up the steps and put headings on them, are you saying to break each step apart from one another with a little whitespace and make them headings or break the steps up into groups of steps and then put headings on each group of steps? 
3. Making the headings match the worked example's headings is a little tricky as the general steps aren't quite 1:1 with the worked example steps with respect to order. For instance, the first step in the general directions that apply to each node is to perform a planned switchover if the target node in question is the current write leader. However, this step is in ### Next Steps of the worked example because the worked example is the upgrade of a shadow node (as it would be weird to start with the write leader in the worked example.